app/util/ContentUpgrade.scala

```diff
@@ -30,14 +30,21 @@ object ContentUpgrade {
 
   def upgradeResponse(json: JValue) = {
     json \ "response" match {
+
       case jsObject: JObject =>
-        JObject("response" -> (jsObject ~ JObject(ContentFields flatMap { field =>
-          jsObject \ field match {
-            case JArray(items) => Some(field -> JArray(items.map(upgradeItem)))
-            case item: JObject => Some(field -> upgradeItem(item))
-            case _ => None
+        val newValues:List[JField] = jsObject.values.map { case (key, _) =>
+          if (ContentFields.contains(key)) {
+            jsObject \ key match {
+              case JArray(items) => (key -> JArray(items.map(upgradeItem)))
+              case item: JObject => (key -> upgradeItem(item))
+              case value => (key, value)
+            }
+          } else {
+            (key, jsObject \ key)
```

Comment:
I expect that `jsObject \ key` here is also the right handside of `(key, _)` in the pattern match?